ThreadPool.h

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright (c) 2021 Chiu Yen-Chen, Swen Sun-Yen, Wen Yong-Wei, Yuan Wei-Chen.
+All rights reserved.
+Use of this source code is governed by a BSD-style license that can be
+found in the LICENSE file. See the AUTHORS file for names of contributors.
+ */
+#ifndef SHORTLINK_SRC_THREADPOOL_H_
+#define SHORTLINK_SRC_THREADPOOL_H_
+#include<vector>
+#include<Thread>
+
+namespace shortlink
+{
+    //threah pool will create many threah at server inital
+    //this class will handle thread operator
+    //Example: 
+    // ThreadPool threadpool = ThreadPool();
+    // threadPool SetPoolSize(100);
+    class ThreadPool
+    {
+    private:
+        //maxima threads in this thread
+        int maxPoolSize;
+        //current threads in this thread pool
+        int curPoolSize;
+        //thread pool
+        std::vector<thread> threads;
+
+    public:
+        //return how many unused thread in this class
+        int GetFreePoolSize() const;
+        //create more thread 
+        void AddFreePool(const int& PoolSize);
```

Comment:
please use underscore for naming. `pool_size`
https://google.github.io/styleguide/cppguide.html#Variable_Names